tests/test_client_helpers.py

```diff
@@ -0,0 +1,107 @@
+import random
+import string
+import uuid
+
+import pytest
+
+import lakefs_spec.client_helpers as client_helpers
+from lakefs_spec import LakeFSFileSystem
+from tests.util import RandomFileFactory
+
+
+def test_create_tag(
+    random_file_factory: RandomFileFactory, fs: LakeFSFileSystem, repository: str, temp_branch: str
+) -> None:
+    random_file = random_file_factory.make()
+    lpath = str(random_file)
+    rpath = f"{repository}/{temp_branch}/{random_file.name}"
+    fs.put(lpath, rpath, precheck=False)
+    client_helpers.commit(
+        client=fs.client, repository=repository, branch=temp_branch, message="Commit File Factory"
+    )
+    tag_name = f"Change_{uuid.uuid4()}"
+    client_helpers.create_tag(
+        client=fs.client, repository=repository, ref=temp_branch, tag=tag_name
+    )
+
+    assert any(
+        commit.get("id") == tag_name for commit in fs.client.tags_api.list_tags(repository).results
+    )
```

Comment:
I get why this is necessary - maybe we should add a `temporary_tag` context manager to avoid cluttering the test repo?

Reply:
As per discussion, did not move anything into a fixture as to keep the testing logic in the test. However, to ensure an uncluttered repo, I moved the tag creation into a `try finally` block where the `finally` part takes care of cleanup.